iiblib/iib_build_details_model.py

```diff
@@ -1,135 +1,259 @@
+
+    def _get_args(self, data):
+        attrs = {}
+        for general_attr in self._general_attrs:
+            attrs[general_attr] = data[general_attr]
+        for optional_attr, default_maker in self._optional_attrs.items():
+            if optional_attr in data:
+                attrs[optional_attr] = data[optional_attr]
+            else:
+                attrs[optional_attr] = default_maker()
+        for operation_attr in self._operation_attrs:
+            attrs[operation_attr] = data[operation_attr]
+        return attrs
 
     def to_dict(self):
-        return {
-            "id": self.id,
-            "state": self.state,
-            "state_reason": self.reason,
-            "state_history": self.state_history,
-            "from_index": self.from_index,
-            "from_index_resolved": self.from_index,
-            "bundles": self.bundles,
-            "removed_operators": self.removed_operators,
-            "organization": self.organization,
-            "binary_image": self.binary_image,
-            "binary_image_resolved": self.binary_image_resolved,
-            "index_image": self.index_image,
-            "request_type": self.request_type,
-            "arches": self.arches,
-            "bundle_mapping": self.bundle_mapping,
-            "omps_operator_version": self.omps_operator_version,
-        }
+        return self._data
 
     def __eq__(self, other):
+        return isinstance(other, self.__class__) and self._data == other._data
+
+    def __getattribute__(self, name):
         if (
-            self.id == other.id
-            and self.state == other.state
-            and self.reason == other.reason
-            and self.state_history == other.state_history
-            and self.from_index == other.from_index
-            and self.from_index_resolved == other.from_index_resolved
-            and self.bundles == other.bundles
-            and self.removed_operators == other.removed_operators
-            and self.organization == other.organization
-            and self.binary_image == other.binary_image
-            and self.binary_image_resolved == other.binary_image_resolved
-            and self.index_image == other.index_image
-            and self.request_type == other.request_type
-            and self.arches == other.arches
-            and self.bundle_mapping == other.bundle_mapping
-            and self.omps_operator_version == other.omps_operator_version
+            name in object.__getattribute__(self, "_operation_attrs")
+            or name in object.__getattribute__(self, "_optional_attrs")
+            or name in object.__getattribute__(self, "_general_attrs")
         ):
-            return True
-        return False
+            return object.__getattribute__(self, "_data")[name]
+        else:
+            return object.__getattribute__(self, name)
+
+
+class AddModel(IIBBuildDetailsModel):
+    """
+    AddModel class handling data from builds/add endpoint, and
```

Comment:
@lipoja what do you think about the description?

Reply:
Thanks, I think it is OK.